packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -18,8 +20,12 @@ class SharedPreferences {
   static const String _prefix = 'flutter.';
   static Completer<SharedPreferences> _completer;
 
-  static SharedPreferencesStorePlatform get _store =>
-      SharedPreferencesStorePlatform.instance;
+  static SharedPreferencesStorePlatform get _store {
+    if (LocalPlatform().isWindows) {
```

Comment:
Let's add a comment here referencing https://github.com/flutter/flutter/issues/52267 so it's clear what this is a workaround for.